nvflare/fuel/f3/sfm/heartbeat_monitor.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import time
+from threading import Event, Thread
+from typing import Dict
+
+from nvflare.fuel.f3.comm_config import CommConfigurator
+from nvflare.fuel.f3.drivers.driver_params import DriverCap
+from nvflare.fuel.f3.sfm.constants import Types
+from nvflare.fuel.f3.sfm.sfm_conn import SfmConnection
+
+log = logging.getLogger(__name__)
+
+HEARTBEAT_TICK = 5
+DEFAULT_HEARTBEAT_INTERVAL = 60
+
+
+class HeartbeatMonitor(Thread):
+    def __init__(self, conns: Dict[str, SfmConnection]):
```

Comment:
This requires creating all the connections first, then create the HeartbeatMonitor. Do we have the use case of creating additional connection after the HeartbeatMonitor created? In that case, we need the function like add_connection()?

Reply:
Right now, the HearbeatMonitor is created before all connections. The `conns` will be updated if new connection is created.